### DISCUSSION TITLE
Handling Selection and Click Events for Elements with Blank Inner Areas in JointJS

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2387] [Q&A] [answered]
### Introduction

I am currently working on a project where I have to deal with various shapes on a graph, including circles, ellipses, and lines. These shapes are wrapped in a <g> tag ( lets say rectangle for now ), but they don’t always take up the full width and height of the wrapping element, leaving some blank space inside.(like circle or ellipsis )

In the JointJS library, if we implement the following code
`paper.on("blank:pointerdown", (evt, x, y) => {
            selection.startSelecting(evt);
    });`
This function activates a feature where, if you click and drag the pointer starting from a blank area of the paper and move it onto an element, or if you completely enclose an element within the dragged area, those elements will be selected.

But with the same above code if we click in the black areas inside the wrapping rectangle of the shapes then that results in the element being selected directly without any dragging

However, I am looking for a way to modify this behavior. Specifically, I want to trigger the element click event instead of selecting the element when clicking on the blank area inside the rectangle.

I have attached a screen recording to this discussion which demonstrates the current behavior. I believe this will help in understanding the issue more clearly.

Any guidance or suggestions on how to achieve this would be greatly appreciated. Thank you in advance for your time and assistance.

### Steps to reproduce

https://github.com/clientIO/joint/assets/116190897/95d99208-78cb-4617-be30-12cbbe5a5698

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You can put a transparent rectangle behind the ellipse (a custom shape is required).

**Reply by VADDIJASWANT** [reply]
Hi have a small Idea I don't know if it is possible, In the function
`
 paper.on("blank:pointerdown", (evt, x, y) => {
            selection.startSelecting(evt);
    });
`
when we click on the blank area inside the element wrapper, then there should be an elementView  or something related to the element in the evt variable right so that selection when taking startSelecting it is directly able to select the element when we click that area, so I was thinking is there a way where we can get the elementView from the evt variable and trigger the click event for that element

**Reply by kumilingus** [reply, marked as answer]
Well, you can find the element under the cursor from the `event.target` or the point (`x`,`y`).
```js
const selection = new joint.ui.Selection({
    paper: paperScroller,
    useModelGeometry: true,
});

paper.on('blank:pointerdown', (evt, x, y) => {
    const [element] = paper.model.findModelsFromPoint({ x, y });
    if (element) {
        // const elementView = paper.findViewByModel(element);
        // do something with elementView
        selection.collection.reset([element]);
    } else {
        selection.startSelecting(evt);
    }
});
```
I'm not 100% sure what you're trying to achieve, but currently it's not possible to display the lasso and selection boxes at the same time using `ui.Selection` plugin.

You can implement a custom lasso though as shown here: https://jsfiddle.net/kumilingus/1hx9obuq/
